deepspeed/moe/utils.py

```diff
@@ -150,3 +150,29 @@ def split_params_into_different_moe_groups_for_optimizer(
 
 def is_moe_param_group(param_group):
     return param_group.get('moe', False)
+
+
+def configure_moe_param_groups(model_parameters: List):
+    # peak at the first element to determine how to proceed
+    first = model_parameters[0]
+
+    # match torch.optim.Optimizer expectations
+    if not isinstance(first, (torch.Tensor, dict)):
+        raise TypeError("param argument that would be given to the optimizer should be "
+                        f"an iterable of Tensors or dicts, but got {type(first)}")
+
+    # Case 1: model_parameters is a list of torch.nn.Parameter
```

Comment:
Can we briefly validate that all elements in `model_parameters` are the same class?